src/sentry/static/sentry/app/views/taskList/tasks.jsx

```diff
@@ -50,18 +47,19 @@ class Tasks extends React.Component {
 
       const prunedTask = {count, name, taskDefinitionKey};
 
-      if (!processes[processDefinitionKey]) {
-        processes[processDefinitionKey] = {
-          processDefinitionKey,
-          processDefinitionName,
-          count,
-        };
-        processes[processDefinitionKey].tasks = [prunedTask];
-      } else {
-        processes[processDefinitionKey].count += count;
-        processes[processDefinitionKey].tasks.push(prunedTask);
-      }
-    });
+      r[processDefinitionKey] = r[processDefinitionKey]
+        ? {...r[processDefinitionKey]}
+        : {
+            tasks: [],
+            count: 0,
+            processDefinitionKey,
+            processDefinitionName,
+          };
+      r[processDefinitionKey].count += count;
+      r[processDefinitionKey].tasks.push(prunedTask);
+
+      return r;
+    }, {});
```

Comment:
I was wondering if this could become like just a one-liner with a helper function or library that would in turn use `reduce` (I guess). Similar to what this person is doing: https://stackoverflow.com/a/34890276/282024

Both approaches seem to contain a lot of boilerplate to my javascript-untrained eyes.

Reply:
Not really, that would work if I were creating a simple object of the format 'processkey': [tasks]. But I'm actually creating a nested object like 'processkey': {tasks: [tasks], count: x ...}